plugins/user/cronjobs

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# -*- python -*-
+# -*- coding: utf-8 -*-
+#
+# Plugin to monitor the number of cronjobs running per user
+#
+# Usage: Place in /etc/munin/plugins/ (or link it there using ln -s)
+#   Add this to your /etc/munin/plugin-conf.d/munin-node:
+#       [cronjobs]
+#       user root
+#
+# 2019-09-09: v 1.0 pcy <pcy.ulyssis.org>: created
+#
+# Parameters understood:
+#
+#   config   (required)
+#   autoconf (optional - used by munin-config)
+#
+# magic markers:
+#%# family=auto
+#%# capabilities=autoconf
+
+from datetime import datetime, timedelta
+import shutil
+import sys
+
+
+def isokline(ln: str):
+    if "CRON" not in ln:
+        return None
+    sp = ln.split()
+    return sp if sp[5][0] == '(' and sp[5][-1] == ')' and sp[5] != "(root)" else None
+
+
+def getfilelines():
+    with open('/var/log/syslog', 'r') as f:
+        return [x for x in map(isokline, f.readlines()) if x is not None]
+
+
+def autoconf():
+    print("no (need cron installed)" if shutil.which('cron') is None else "yes")
+
+
+def config():
+    usernames = set(map(lambda e: e[5][1:-1], getfilelines()))
+    print("""\
+graph_title Cron jobs per user
+graph_vlabel jobs
+graph_category processes
+""")
+    for n in usernames:
+        print(n + ".label " + n)
+        print(n + ".info jobs of user " + n)
+        print(n + ".type GAUGE")
+        print(n + ".draw LINE")
+
+
+def fetch():
+    cutoff = timedelta(minutes=5)
```

Comment:
This approach is a bit tricky, since it relies on the munin-node being requested in a static interval (exactly every 5 minutes). It will cause overlaps, if the timing of the munin master request is not perfectly stable (e.g. random order of requesting nodes  or varying timing of processing the plugins *before* this one).

The perl module `Munin::Plugin` offers an `open_tail` function for this purpose. Sadly there is no similar implementation provided for python plugins.

If you cannot find a way around this potential instability, then I would suggest to document this in the plugins perldoc header clearly.

Reply:
For `journald`, it'd be possible to use the `--show-cursor`/`--after-cursor=` arguments, but I don't want to rely on `journalctl` being installed.

Another approach would be storing the timestamp of the last run time in a pickle and writing that to the statefile, and comparing that to the current timestamp in the next run. But that feels a bit dirty, so would that be ok?

Reply:
Yes, using a statefile would be a suitable approach in my opinion. I would suggest to just store the timestamp (without pickling) or using a json format, if you feel like.

This would indeed solve the fragility, that I felt when reading the plugin.

The rest of the plugin looks great after your latest changes. Thanks!

Reply:
Added the statefile now. I wanted to use pickle or sth instead because parsing/serializing datetimes to strings has proven to be annoying once again. But it's done this way instead now.